Error running deploy command in 11.10.0 - "Error: Cannot find module 'is-npm'"

<!-- DO NOT DELETE
validate_template=true
template_path=.github/ISSUE_TEMPLATE/bug_report.md
-->

<!--
Thank you for contributing to the Firebase community!

Think you found a bug?
=======================
Yeah, we're definitely not perfect! Please use this template and include a minimal repro when opening the issue. If you know how to solve the issue, please create a Pull Request, and we'd be happy to review it!

Have a feature request?
========================
Great, we love hearing how we can improve our products! However, do not use this template to submit a feature request. Please submit your feature requests to: https://firebase.google.com/support/contact/bugs-features/

Have a usage question?
=======================
We get lots of those and we love helping you, but GitHub is not the best place for them and they will be closed. Please take a look at the guide first: https://firebase.google.com/docs/cli/

If the official documentation doesn't help, try asking through our official support channel: https://firebase.google.com/support/

Additional locations to check for solutions or assistance from the community:
- Stack Overflow: https://stackoverflow.com/
- Firebase Slack Community: https://firebase.community/

*Please avoid duplicate posting across multiple channels!*
-->

### Environment info

<!-- What version of the Firebase CLI (`firebase-tools`) are you using? Note that your issue may already be fixed in the latest versions. The latest version can be found at https://github.com/firebase/firebase-tools/releases -->

<!-- Output of `firebase --version` -->

**firebase-tools:**


```
➜ npx firebase-tools --version
Downloading Chromium r869685 - 112.4 Mb [====================] 100% 0.0s 
11.10.0
Cannot find module 'is-npm'
Require stack:
- /Users/stewart/.npm/_npx/45995/lib/node_modules/firebase-tools/node_modules/update-notifier/index.js
- /Users/stewart/.npm/_npx/45995/lib/node_modules/firebase-tools/lib/bin/firebase.js
```

<!-- e.g. macOS, Windows, Ubuntu -->

**Platform:**

macOS (m1)

### Test case

<!-- Provide a minimal, complete, and verifiable example (http://stackoverflow.com/help/mcve) -->
Using version 11.10 to do a `hosting:channel:deploy` does succeed in deploying, however it fails to exit successfully as a error is thrown `Cannot find module 'is-npm'`

### Steps to reproduce

<!-- Provide the steps needed to reproduce the issue with the above test case. -->

run `npx firebase-tools hosting:channel:deploy pr414-blah --only special-deploy --expires 14d --project test-project  --debug`

### Expected behavior

<!-- What is the expected behavior? -->
The deploy succeeds and exits successfully with no errors.

### Actual behavior

<!-- Run the command with --debug flag, and include the logs below. -->
```
❯ npx firebase-tools hosting:channel:deploy pr414-blah --only special-deploy --expires 14d --project test-project  --debug
Downloading Chromium r869685 - 112.4 Mb [====================] 100% 0.0s 
[2022-09-23T08:52:38.718Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[2022-09-23T08:52:38.719Z] > authorizing via signed-in user (***@***.com)
[2022-09-23T08:52:38.719Z] [iam] checking project test-project for permissions ["firebase.projects.get","firebasehosting.sites.update"]
[2022-09-23T08:52:38.721Z] > refreshing access token with scopes: []
...
✔  hosting:channel: Channel URL (***): https://***.web.app [expires 2022-10-07 10:52:45] 
[2022-09-23T08:53:10.615Z] Error: Cannot find module 'is-npm'
Require stack:
- /Users/stewart/.npm/_npx/46162/lib/node_modules/firebase-tools/node_modules/update-notifier/index.js
- /Users/stewart/.npm/_npx/46162/lib/node_modules/firebase-tools/lib/bin/firebase.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:101:18)
    at lazy (/Users/stewart/.npm/_npx/46162/lib/node_modules/firebase-tools/node_modules/import-lazy/index.js:2:51)
    at /Users/stewart/.npm/_npx/46162/lib/node_modules/firebase-tools/node_modules/import-lazy/index.js:10:11
    at UpdateNotifier.notify (/Users/stewart/.npm/_npx/46162/lib/node_modules/firebase-tools/node_modules/update-notifier/index.js:117:59)
    at process.<anonymous> (/Users/stewart/.npm/_npx/46162/lib/node_modules/firebase-tools/lib/bin/firebase.js:108:20)
    at process.emit (events.js:412:35)
    at process.exit (internal/process/per_thread.js:179:15)
    at /Users/stewart/.npm/_npx/46162/lib/node_modules/firebase-tools/lib/command.js:105:25
    at processTicksAndRejections (internal/process/task_queues.js:95:5)

Error: An unexpected error has occurred.

```

Note that even  `--version` triggers the error (see at the top of this issue)

We also use ```FirebaseExtended/action-hosting-deploy@v0``` and have this error Error: Cannot find module 'is-npm’ even npm I -D is-npm doesn’t help. Thanks

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,1 +1,2 @@
 - Adds support for codebases in `firebase init functions` flow (#4965).
+- Fixes errors on exit related to `update-notifier` (#5010).

--- a/src/bin/firebase.ts
+++ b/src/bin/firebase.ts
@@ -127,18 +127,29 @@ process.on("exit", (code) => {
   }
 
   // Notify about updates right before process exit.
-  const updateMessage =
-    `Update available ${clc.gray("{currentVersion}")} → ${clc.green("{latestVersion}")}\n` +
-    `To update to the latest version using npm, run\n${clc.cyan(
-      "npm install -g firebase-tools"
-    )}\n` +
-    `For other CLI management options, visit the ${marked(
-      "[CLI documentation](https://firebase.google.com/docs/cli#update-cli)"
-    )}`;
-  // `defer: true` would interfere with commands that perform tasks (emulators etc.)
-  // before exit since it installs a SIGINT handler that immediately exits. See:
-  // https://github.com/firebase/firebase-tools/issues/4981
-  updateNotifier.notify({ defer: false, isGlobal: true, message: updateMessage });
+  try {
+    const updateMessage =
+      `Update available ${clc.gray("{currentVersion}")} → ${clc.green("{latestVersion}")}\n` +
+      `To update to the latest version using npm, run\n${clc.cyan(
+        "npm install -g firebase-tools"
+      )}\n` +
+      `For other CLI management options, visit the ${marked(
+        "[CLI documentation](https://firebase.google.com/docs/cli#update-cli)"
+      )}`;
+    // `defer: true` would interfere with commands that perform tasks (emulators etc.)
+    // before exit since it installs a SIGINT handler that immediately exits. See:
+    // https://github.com/firebase/firebase-tools/issues/4981
+    updateNotifier.notify({ defer: false, isGlobal: true, message: updateMessage });
+  } catch (err) {
+    // This is not a fatal error -- let's debug log, swallow, and exit cleanly.
+    logger.debug("Error when notifying about new CLI updates:");
+    if (err instanceof Error) {
+      logger.debug(err);
+    } else {
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      logger.debug(`${err}`);
+    }
+  }
 });
 
 process.on("uncaughtException", (err) => {

